### DISCUSSION TITLE
Add instances from an imported xml file

### DISCUSSION THREAD

**Discussion by herculesp17** [FreeOpcUa/opcua-asyncio#1262] [General]
I have no clue how to add new instances on a server, having already imported the xml files I need. 
I’ve been looking around and found some module called node management service, but I can’t import it. Is it an old module? I don’t find it in the docs either. Maybe someone can help me out! 
thanks in advance

**Comment by schroeder-** [comment]
If you imported an xml and it's just a description, for example companion specifications, you start and instantiate object and variables you need. [See example](https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/server-instantiate-object.py)

If you imported an xml with a complete addresspace, your job is to connect the create variables/objects/methods to your I/O source or logic.

**Reply by herculesp17** [reply]
> If you imported an xml with a complete addresspace, your job is to connect the create variables/objects/methods to your I/O source or logic.

Can you give me an example how exactly I'm doing that?
